### PR TITLE
Pagination, tag, image upload

### PR DESCRIPTION
**Pull request by aqfaridi**
**What this PR does -** 
- Featured Image for Article and Profile Picture of User
- Pagination for the articles
- Tagging for articles has been added

**Dependency :** 
- will_paginate gem
- acts-as-taggable-on gem
- paperclip gem

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -29,6 +29,11 @@ gem 'haml-rails', '~> 0.9.0'
 gem 'devise', '~> 4.1.1'
 gem 'annotate', '~> 2.7.1'
 gem 'mail', '~> 2.6.4'
+gem 'will_paginate', '~> 3.1.0'
+gem 'kaminari'
+gem 'acts-as-taggable-on', '~> 3.4'
```
Comment (coolRaider) [inline Gemfile:33]: please add comments with reason for using a particular gem

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -13,5 +13,8 @@
 //= require jquery
 //= require jquery_ujs
 //= require twitter/bootstrap
+//= require bootstrap-tagsinput
 //= require turbolinks
 //= require_tree .
+
+
```
Comment (coolRaider) [inline app/assets/javascripts/application.js]: please remove multiple newline at the end of files

File: app/controllers/tags_controller.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  end
+
+  def show
+    @tag =  ActsAsTaggableOn::Tag.find_by_name(params[:id])
+    if @tag == nil 
+    	@tag =  ActsAsTaggableOn::Tag.find(params[:id])
+    end
+    @articles = Article.tagged_with(@tag.name)
+  end
+end
```
Comment (coolRaider) [inline app/controllers/tags_controller.rb:10]: please add one newline at end of file

**Comment by coolRaider** [comment]
Try to split the code you push into multiple commits so parts of it could be reverted if need be.

File: app/controllers/tags_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class TagsController < ApplicationController
+  def index
+    @tags = ActsAsTaggableOn::Tag.all
+  end
+
+  def show
+    @tag =  ActsAsTaggableOn::Tag.find_by_name(params[:id])
```
Comment (tarunjadhwani) [inline app/controllers/tags_controller.rb]: pass in name here.

Comment (tarunjadhwani) [inline app/controllers/tags_controller.rb]: or send id and optimise this.

**Pull request merged**